src/main/java/com/cargohub/services/impl/UserServiceImpl.java

```diff
@@ -99,6 +103,38 @@ public UserDto updateUser(long id, UserDto user) {
         return modelMapper.map(storedUser, UserDto.class);
     }
 
+    @Override
+    public boolean updateUserPassword(long id, String password) {
+        UserEntity userEntity = getUserEntityById(id);
+        userEntity.setEncryptedPassword(bCryptPasswordEncoder.encode(password));
+        userRepository.save(userEntity);
+        return true;
+    }
+
+    @Override
+    public boolean addUsersBillingDetailsCard(long id, BillingDetailsDto billingDetailsDto) {
+        UserEntity userEntity = getUserEntityById(id);
+        BillingDetailsEntity billingDetailsEntity = modelMapper.map(billingDetailsDto, BillingDetailsEntity.class);
+        userEntity.getBillingDetails().add(billingDetailsEntity);
+        billingDetailsEntity.setUserDetails(userEntity);
+        userRepository.save(userEntity);
+        return true;
```

Comment:
before returning result, make sure that userEntity is saved

Reply:
done, please check